lib/dal.js

```diff
@@ -2,35 +2,60 @@ var Q = require('q');
 var fetchJSON = require('./fetchJSON.js');
 var customError = require('./customError.js');
 
+/**                                                                              
+ * Represents an error that has both a name and message.
+ * @param {String} name - the name/type of the error
+ * @param {String} message - the corresponding message
+ * Mainly used to avoid crashing the program, as it does with regular errors.
+ */
+function CustomError(name, message) {
+  this.name = name;
+  this.message = message;
+}
+
 /** 
  * Attempts to access configuration file data for the shared extension
+ *   If this is the first time accessing this said extension, then it creates
+ *   a new sharedExtension "object" and its relevant underlying data structures,
+ *   adds it to the array of extsInUse and returns it.
+ *   Simply returns the object in extsInUse if already previously created. This
```

Comment:
This comment is somewhat awkward. I'd rephrase it slightly:
"If this is the first time accessing this extension, then ... . Otherwise, it returns the previously created object."